src/roadstop.cpp

```diff
@@ -291,8 +298,13 @@ void RoadStop::Entry::Enter(const RoadVehicle *rv)
 {
 	/* we cannot assert on this->occupied < this->length because of the
 	 * remote possibility that RVs are running through each other when
-	 * trying to prevention an infinite jam. */
+	 * trying to prevention an infinite jam,
+	 * or if an articulated vehicle, has partially entered the stop.
+	 */
+	this->vehicles += 1;
 	this->occupied += rv->gcache.cached_total_length;
+	this->occupied = std::min(this->occupied, this->length);
```

Comment:
This, along with the modified comment about articulated vehicles, feels like a separate bugfix to clamp the occupied length to the roadstop length. I don't see how it relates to this PR. If so, it should be a separate PR.

Reply:
Right — because it affects the current code, I think changing it should be a separate PR.